beer-garden/beer-garden
# Issue/1067

Closes #1067 

## Test Instructions
1. There are integration tests that exercise the added functionality. Check the code for the tests at `test/integration/scheduler/jobs/job_import_export_test.py` and see the successful run of those tests [here](https://github.com/beer-garden/beer-garden/actions/runs/1113384087).
2. Alternately, the functionality can be tested through `curl` with the following:

* If none exist on the system, create a scheduled job through the BG UI in the normal way. To find the IDs of the jobs on the system, the easiest way seems to be to use Swagger's `/api/v1/jobs` section, which will list all the jobs. Note the ID of a least one job.
* From the command line, type

    ```bash
    curl -X 'POST'  'http://localhost:8080/api/v1/export/jobs' \
    -H 'accept: application/json' -H 'Content-Type: application/json' \
    -d '{
    "ids": [
    <the job ID you noted including the double quotes>
     ]
    }'
    ```

* Verify the returned JSON is accurate for the job exported. Save the returned JSON in a convenient location for the next test.
* Type at the command line:

    ```bash
    curl -X 'POST' 'http://localhost:8080/api/v1/import/jobs' \
    -H 'accept: application/json' \
    -H 'Content-Type: application/json' \
    -d'<the body of the JSON returned from the last step including the enclosing braces>'
    ```

* Verify that the job has been created through the "Requests" page of the BG UI and that it is a copy of the exported job. Note that the "Next Run Time" field will be blank until the scheduler picks up the new job.

NB: The job export functionality can also be checked using the Swagger UI. However, due to what appears to be a bug in this version of Swagger, the JSON returned in the web page is wrong. Swagger will not accept the date-time objects unless they are enclosed in quotes, but BG will then reject them.
